examples/nlp/lasertagger/official_lasertagger/sari_hook.py

```diff
@@ -0,0 +1,169 @@
+# coding=utf-8
+# Copyright 2019 The Google Research Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""SARI score for evaluating paraphrasing and other text generation models.
+
+The score is introduced in the following paper:
+
+   Optimizing Statistical Machine Translation for Text Simplification
+   Wei Xu, Courtney Napoles, Ellie Pavlick, Quanze Chen and Chris Callison-Burch
+   In Transactions of the Association for Computational Linguistics (TACL) 2015
+   http://cs.jhu.edu/~napoles/res/tacl2016-optimizing.pdf
+
+This implementation has two differences with the GitHub [1] implementation:
+  (1) Define 0/0=1 instead of 0 to give higher scores for predictions that match
+      a target exactly.
+  (2) Fix an alleged bug [2] in the deletion score computation.
+
+[1] https://github.com/cocoxu/simplification/blob/master/SARI.py
+    (commit 0210f15)
+[2] https://github.com/cocoxu/simplification/issues/6
+"""
```

Comment:
is it port? do we have SWIPAT for that?

Reply:
Because it is reimplemented bottom-up here, I didn't apply for SWIPAT to use SARI.

Reply:
I do not understand... what does it mean: bottom-up?

Reply:
I meant it is a reimplementation of the https://github.com/cocoxu/simplification/blob/master/SARI.py